storm-core/src/jvm/org/apache/storm/utils/WorkerBackpressureThread.java

```diff
@@ -43,17 +48,33 @@ static public void notifyBackpressureChecker(Object trigger) {
         }
     }
 
+    public void terminate() {
+        running = false;
+    }
```

Comment:
We could then ignore the interrupted exception in run as it would just be used as a trigger to wake up the running thread.